source/common/common/enum_to_int.h

```diff
@@ -6,5 +6,10 @@ namespace Envoy {
 /**
  * Convert an int based enum to an int.
  */
-template <typename T> uint32_t enumToInt(T val) { return static_cast<uint32_t>(val); }
+template <typename T> constexpr uint32_t enumToInt(T val) { return static_cast<uint32_t>(val); }
+
+/**
+ * Convert an int based enum to a signed int.
+ */
+template <typename T> constexpr int32_t enumToIntSigned(T val) { return static_cast<int32_t>(val); }
```

Comment:
sorry, super nitty nit, but would you mind making this enumToSignedInt?

That's my one request for the non-extension file, so as long as that's fixed I'm fine with @snowp  merging

Reply:
yep yep, on it. Thank you @alyssawilk !